core/src/main/java/org/infinispan/remoting/transport/ValidSingleResponseCollector.java

```diff
@@ -0,0 +1,46 @@
+package org.infinispan.remoting.transport;
+
+import org.infinispan.remoting.RpcException;
+import org.infinispan.remoting.responses.CacheNotFoundResponse;
+import org.infinispan.remoting.responses.ExceptionResponse;
+import org.infinispan.remoting.responses.Response;
+import org.infinispan.remoting.responses.ValidResponse;
+import org.infinispan.util.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+/**
+ * @author Dan Berindei
+ * @since 9.1
+ */
+public abstract class ValidSingleResponseCollector<T> implements ResponseCollector<T> {
```

Comment:
I'm not sure if this class is really useful since all subclasses could use ValidResponseCollector to achieve the same logic.